### PR TITLE
Avoid ItemStack overwriting due to spell casts

### PR DESCRIPTION
**Pull request by P0keDev**
This PR doesn't add any super important functionality, but it was a nice addition to make while also transferring the `ItemStackTransformer` into an internal feature.

Essentially, Wynncraft achieves the small spell cost pop-up that appears when you cast a spell by overwriting the name of your currently held weapon. This presents a small problem because this resets the item in the player's inventory, and with its temporary name our item logic no longer functions. It also looks a little bit silly if you decide to look at your item right after using a spell. `FixSpellOverwriteFeature` solves this by stopping the renamed item from being set in the inventory slot, but uses the item's data to still display the appropriate spell pop-up. This keeps the player's item clean and doesn't remove any Wynncraft functionality.

Additionally, I changed any `EventFactory` methods with return values to return the event itself, instead of just the boolean canceled value. I think this is a good pattern to follow (and can be done in one line, with the changes to `post`), since many mixins will require more information from the packet than the cancellation status - including the `SlotMixin` here. I also reorganized all the methods in `EventFactory` because god damn it was a mess

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/internal/ItemStackTransformerFeature.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.function.Function;
+import java.util.function.Predicate;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@EventListener
+public class ItemStackTransformerFeature extends InternalFeature {
+
+    private final Map<Predicate<ItemStack>, Function<ItemStack, WynnItemStack>> TRANSFORMERS = new HashMap<>();
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/internal/ItemStackTransformerFeature.java:28]: This could be made a functional interface

**Review by Incompleteusern — APPROVED** [review]
LGTM

File: common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    @SubscribeEvent
+    public void onSetSlot(SetSlotEvent event) {
+        ItemStack item = event.getItem();
+        if (item == null || item.isEmpty()) return;
+
+        Matcher spellMatcher = SPELL_CAST.matcher(item.getHoverName().getString());
+        if (!spellMatcher.find()) return;
+
+        event.setCanceled(true);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java:35]: This should not really be canceled in the middle of a method

File: common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        overwriteHighlightTimer = 40;
+        overwriteHighlightItem = item;
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java:37]: I don’t really understand what this does. Doesn’t wynncraft natively send the packets to cause a highlight timer as well? What happens to the display of spell cast?

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java:37]: Oh wait never mind.

File: common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java (hunk 2 above)
Comment (P0keDev) [reply common/src/main/java/com/wynntils/features/internal/FixSpellOverwriteFeature.java:35]: I don't see why it matters? It's just setting the flag, the actual cancellation doesn't happen until all the the event methods have finished processing

**Review by kristofbolyai — APPROVED** [review]

**Comment by magicus** [comment]
What is the principle you followed when reorganizing EventFactory? Alphabetical?

**Comment by P0keDev** [comment]
I did it by category, and I also included comments that create collapse-able regions in intellij:

**Pull request merged**